phoenix-core/src/it/java/org/apache/phoenix/end2end/SCNIT.java

```diff
@@ -82,28 +85,58 @@ public void testReadBeforeDelete() throws Exception {
 
 	@Test
 	public void testSCNWithTTL() throws Exception {
+		int ttl = 2;
+		String fullTableName = createTableWithTTL(ttl);
+		//sleep for one second longer than ttl
+		Thread.sleep(ttl * 1000 + 1000);
+		Properties props = new Properties();
+		props.setProperty(PhoenixRuntime.CURRENT_SCN_ATTRIB,
+				Long.toString(EnvironmentEdgeManager.currentTime() - 1000));
+		try (Connection connscn = DriverManager.getConnection(getUrl(), props)) {
+			ResultSet rs = connscn.createStatement().executeQuery("select * from " + fullTableName);
+			assertFalse(rs.next());
+			rs.close();
+		}
+	}
+
+	@Test
+	public void testTooLowSCNWithTTL() throws Exception {
+		//if scn is for an older time than a table's ttl, it should throw a SQLException
+		int ttl = 2;
+		String fullTableName = createTableWithTTL(ttl);
+		//need to sleep long enough for the SCN to still find the syscat row for the table
+		Thread.sleep(ttl * 1000 + 1000);
+		Properties props = new Properties();
+		props.setProperty(PhoenixRuntime.CURRENT_SCN_ATTRIB,
+			Long.toString(EnvironmentEdgeManager.currentTime() - (ttl + 1)* 1000));
```

Comment:
nit: (ttl+1)*1000 can be a new variable and reused at 108 and 111

Reply:
Good catch, done.